doc/design/gan_api.md

```diff
@@ -0,0 +1,193 @@
+# Design for GAN
+
+GAN (General Adversarial Net) is an important model for unsupervised learning and widely used in many areas. 
```

Comment:
maybe add a link to the official definition

Reply:
Thanks. Done.